### ISSUE TITLE
[4.2] PLG_SYSTEM_REDIRECT_ERROR_UPDATING_DATABASE not translated

### ISSUE THREAD

**Issue by ReLater** [joomla/joomla-cms#39389]
This find came up in a thread in German forum about another issue with `plg_system_redirect`.

### Steps to reproduce not-translated issue
- Activate plugin `plg_system_redirect` with activated setting `collect_urls`.
- "Hack" file `plugins\system\redirect\redirect.php`:
  - In [line 257](https://github.com/joomla/joomla-cms/blob/4.2-dev/plugins/system/redirect/redirect.php#L257) add line: `echo Text::_('PLG_SYSTEM_REDIRECT_ERROR_UPDATING_DATABASE');`
  - Frontend: Enter a wrong url to be redirected to the error page,
  - Output at top of page `PLG_SYSTEM_REDIRECT_ERROR_UPDATING_DATABASE`

### Expected result
`PLG_SYSTEM_REDIRECT_ERROR_UPDATING_DATABASE` translated to `An error occurred while updating the database.`

### Actual result
`PLG_SYSTEM_REDIRECT_ERROR_UPDATING_DATABASE` not translated

### Additional comments
When I set in [line 42](https://github.com/joomla/joomla-cms/blob/4.2-dev/plugins/system/redirect/redirect.php#L42) `protected $autoloadLanguage = true;` instead of `protected $autoloadLanguage = false;` the string is translated.

Question before I provide a PR is if it is by purpose that `$autoloadLanguage` is `false` and not `true`?

**Comment by SharkyKZ** [comment]
>Question before I provide a PR is if it is by purpose that $autoloadLanguage is false and not true?

Yes, because there is no need to load language files on every request.

**Comment by ReLater** [comment]
> Yes, because there is no need to load language files on every request.

Where should the language then be loaded?

**Comment by SharkyKZ** [comment]
Preferably right before the strings are used.

**Comment by ReLater** [comment]
Then I'm out concerning pr and someone else has to take over.
